### PR TITLE
ISSUE-933: remove array_scalar_arithmetic_op for array operator

### LINKED ISSUE #933: [datavalues] remove array scalar arithmetic
**Summary**

Change `DataValueArithmeticOperator::Div` and `DataValueArithmeticOperator::Modulo` from scalar to array vector:
https://github.com/datafuselabs/datafuse/blob/8dbb10f0b5801e59deea7b2c5bcbd233aeca4fcd/common/datavalues/src/data_array_arithmetic.rs#L100-L130

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

* Change `Div` and `Modulo` from scalar to array operation
* Remove array_scalar_arithmetic_op

## Changelog

- Improvement

## Related Issues

Fixes #933

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/datavalues/src/data_array_arithmetic.rs [hunk 1]
```diff
@@ -51,7 +51,8 @@ impl DataArrayArithmetic {
                         scalar.to_array_with_size(1)?
                     }
                 };
-                let result = Self::array_scalar_arithmetic_op(op, &left_array, right_value)?;
+                let right_array = right_value.to_array_with_size(left_array.len())?;
```
Comment (sundy-li) [inline common/datavalues/src/data_array_arithmetic.rs:54]: This only works in the case of `constant % array`.

what about `constant +/-/*  array`?

**Review by sundy-li — CHANGES_REQUESTED** [review]

File: common/datavalues/src/data_array_arithmetic.rs (hunk 1 above)
Comment (sundy-li) [inline common/datavalues/src/data_array_arithmetic.rs:54]: What's the reason to remove `array_scalar_op` ?

It could have a performance rise with this op.

Comment (sundy-li) [reply common/datavalues/src/data_array_arithmetic.rs:54]: From arrow
```
/// SIMD vectorized version of `divide_scalar`.
#[cfg(feature = "simd")]
fn simd_divide_scalar<T>(
```

Comment (BohuTANG) [reply common/datavalues/src/data_array_arithmetic.rs:54]: Aha,I missed `simd_*_scalar`.
From my perf top, the `divide_scalar` function is second performance killer(top 1 is concat_row_to_one_key)
This pull request can be closed now.
Thank you @sundy-li.

Comment (sundy-li) [reply common/datavalues/src/data_array_arithmetic.rs:54]: Ok, I was refactoring the concat_row_to_one_key with vec_hash in #819.

See: https://github.com/datafuselabs/datafuse/blob/f9cbff7a625274c235e355ee70908f9404e3ef48/common/datablocks/src/kernels/data_block_group_by.rs#L76-L87

Comment (BohuTANG) [reply common/datavalues/src/data_array_arithmetic.rs:54]: :100:

**Pull request closed without merging**